### PR TITLE
Implement WriteBuffer float, double and BigInteger functions

### PR DESCRIPTION
**Pull request by hutcheb**
Initial commit, hasn't been tested at all.

### CODE REVIEW

File: plc4j/drivers/knxnetip/src/main/java/org/apache/plc4x/java/knxnetip/utils/KnxHelper.java [hunk 1]
```diff
@@ -46,9 +46,16 @@ public static void f16toBytes(WriteBuffer io, Object param) {
             boolean negative = value < 0;
             final int exponent = Math.getExponent(value);
             final double mantissa = value / Math.pow(2, exponent);
+            //io.writeBit(negative);
+            //io.writeInt(4, exponent);
+            //io.writeDouble(11, Math.getExponent(mantissa), mantissa); //Don't think this works BH
+
+            String mantissaString = Double.toString(mantissa);
+            int mantissaInteger = Integer.parseInt(mantissaString.substring(mantissaString.indexOf('.')));
```
Comment (chrisdutz) [inline plc4j/drivers/knxnetip/src/main/java/org/apache/plc4x/java/knxnetip/utils/KnxHelper.java:54]: What's the String conversion for? I think in this case a simple double-to-int-cast should do the trick ....

**Review by chrisdutz — APPROVED** [review]
LGTM :-) Thanks

File: plc4j/drivers/knxnetip/src/main/java/org/apache/plc4x/java/knxnetip/utils/KnxHelper.java (hunk 1 above)
Comment (hutcheb) [reply plc4j/drivers/knxnetip/src/main/java/org/apache/plc4x/java/knxnetip/utils/KnxHelper.java:54]: The string conversion is to get the digits after the decimal place of the mantissa as part of writing a 16-bit float.
(In my head the mantissa is always going to have a one before the decimal place which can be discarded)
The digits after are then converted to an integer to be written as the manitssa part.

I haven't tested this part of it, it has a slight chance of working for most cases but not all. I'll create a 16-bit floating point PlcValue at some point and test it with Modbus and then move it over to the KNX protocol as a custom PlcValue.

**Comment by hutcheb** [comment]
Little Endian BigInteger functions havn't been tested.

**Pull request closed without merging**